lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -220,6 +227,12 @@ defmodule Kernel.LexicalTracker do
 
   def handle_cast({:add_alias, module, meta, warn}, state) do
     if warn do
+      state =
+        case state do
+          %{aliases: %{^module => meta}} -> put_in(state.aliases[{:shadowed, module}], meta)
```

Comment:
Note: I reused the aliases map since it's only accessed within this file and it is a rare edge-case anyway, but could also have a `shadowed_aliases` key in the global state.